### ISSUE TITLE
Serve images as JPEG2000 for Safari

### ISSUE THREAD

**Issue by widersky** [rosell-dk/webp-express#294]
As you might know, Safari doesn't support WebP for now. Did you consider to add JPEG2000 support to plugin?

**Comment by kropkikreski** [comment]
Yes! We would love to see JPEG200 support for iOS and safari on mac.

**Comment by rosell-dk** [comment]
It is a good idea. But it requires a lot of work. And perhaps the plugin should be renamed as well.  "NextGen image formats" springs to mind.

However, converting to jp2 with PHP is not well supported. It can be done by executing the Imagemagick binary. The Imagick PHP extension however does not seem to support it and neither does Gd or Vips. So the feature will only work out of the box on few systems.

I could add a new conversion method which executes the [open jpeg](http://www.openjpeg.org/) binary as suggested [here]( https://www.useragentman.com/blog/2015/01/14/using-webp-jpeg2000-jpegxr-apng-now-with-picturefill-and-modernizr/). But that method will also be based on executing a binary, which I suspect is restricted on most systems.

It seems that in order to have such conversions available to more than a few will require setting up a cloud conversion service. But as I cannot afford to provide such service for free, the task of implementing such would be significant. But perhaps we can find an existing one to connect to.

For the .htaccess, I could use conditional rewrites with the condition:
`RewriteCond %{HTTP_USER_AGENT} Safari` [similar to here](http://www.htaccesstools.com/articles/detect-and-redirect-iphone/)
